services/files/run_pull.go

```diff
@@ -30,8 +32,21 @@ type PullParams struct {
 	Retrieve  string
 }
 
+func (p PullParams) validate() error {
+	if p.URI == "" && !p.All {
+		return fmt.Errorf("uri or --all is required")
```

Comment:
Not sure how the final concatenated string will look like. But want to make it clear

```suggestion
		return fmt.Errorf("either uri or --all is required")
```
or
```suggestion
		return fmt.Errorf("provide a file pattern or use the --all flag")
```